s3path.py

```diff
@@ -1192,7 +1205,71 @@ def get_presigned_url(self, expire_in: Union[timedelta, int] = 3600) -> str:
         return self._accessor.get_presigned_url(self, expire_in)
 
 
-class StatResult(namedtuple('BaseStatResult', 'size, last_modified')):
+class VersionedS3Path(S3Path):
+
+    def __new__(
+        cls, *args: Union[str, Path], version_id: Optional[str] = None
+    ) -> Union[S3Path, 'VersionedS3Path']:
+        if version_id is None:
+            return S3Path(*args)
+        else:
+            return super().__new__(cls, *args)
+
+    def __init__(self, *args: Union[str, Path], version_id: str) -> None:
```

Comment:
thanks for the suggestion. I used most of you suggestion but I did not include the `__rtruediv__` because I think it makes more sense that relative paths are not allowed for versioned paths. let me know if you disagree.

Reply:
Hi @nlangellier 
Why do you don't want to support relative paths?
what do you think about this use case for example:
``` python
bucket = VersionedS3Path('/bucket/')
for file_name in ['f0', 'f1', ..., 'fn']:
    path = bucket / VersionedS3Path(file_name, version_id=...)
    ...
```
according to pathlib this is a valid api
In S3Path we check if the path is absolute only in the level of methods that do api calls for AWS
So all methods in Pure classes doesn't really need this check

Reply:
yes you are right. that should be allowed. i added both `__truediv__` and `__rtruediv__` methods to properly handle all possible combinations:
* `VersionedS3Path` / `VersionedS3Path`
* `VersionedS3Path` / `S3Path`
* `VersionedS3Path` / `str`
* `S3Path` / `VersionedS3Path`
* `str` / `VersionedS3Path`